autoscheduler/user_sessions/views.py

```diff
@@ -1,22 +1,61 @@
+import json
 from rest_framework.response import Response
-from rest_framework.decorators import api_view
+from rest_framework.decorators import api_view, parser_classes
+from rest_framework.parsers import JSONParser
 
 @api_view(['GET'])
 def get_last_term(request):
-    """ View that returns JSON containing last term for the current user's session.
-        Used by landing page to determine whether to redirect to schedules. """
-    term = request.session.setdefault('term', '')
-    response = {'term': term}
+    """ API endpoint that returns JSON containing last term for the user's session. """
+    term = request.session.get('term')
+    response = {}
+    if term:
+        response['term'] = term
     return Response(response)
 
 @api_view(['PUT'])
 def set_last_term(request):
-    """ View that sets term for the current session. Called when a term is selected,
-        or when the title bar is clicked to unset last term.
+    """ API endpoint that sets term for the current session. Called when a term is
+        selected, or when the title bar is clicked to unset last term.
     """
     term = request.query_params.get('term')
     # empty string term is valid (used to unset term), so explicitly check for None
     if term is None:
         return Response(status=400)
     request.session['term'] = term
     return Response()
+
+@api_view(['PUT'])
+@parser_classes([JSONParser])
+def save_courses(request):
+    """ API endpoint that saves course cards for user's current term. """
```

Comment:
I think it would be worth mentioning that this/both route(s) can store any format of data, since really the only thing related to courses is that it saves the data with the `courses` key in the session.

Reply:
Yeah I put a comment about it in a test instead of the actual API for some reason, will change